command/agent/job_endpoint.go

```diff
@@ -704,6 +704,13 @@ func ApiTaskToStructsTask(apiTask *api.Task, structsTask *structs.Task) {
 						Header:        check.Header,
 						Method:        check.Method,
 					}
+					if check.CheckRestart != nil {
+						structsTask.Services[i].Checks[j].CheckRestart = &structs.CheckRestart{
```

Comment:
Capture this in the test in job_endpoint_test.go

Reply:
Added https://github.com/hashicorp/nomad/pull/3105/commits/924813d57e06ca5bf6510c998a950f3118900fc7